source/class/qx/data/SingleValueBinding.js

```diff
@@ -857,6 +857,7 @@ qx.Class.define("qx.data.SingleValueBinding", {
      */
     __setTargetValue(targetObject, targetPropertyChain, value) {
       // get the last target object of the chain
+      var result;
```

Comment:
what's point to declare variable far from where it is used?

Reply:
I much prefer all variables declared at the top of the block to which they apply, so that it's not necessary to search for a declaration; you always know where to look. This is particularly true with `var` (vs. `let`), where I prefer all variables declared at the top of the function which ends up happening by "hoisting" anyway.